sw/host/opentitantool/src/command/image.rs

```diff
@@ -233,11 +233,19 @@ impl CommandDispatch for ManifestUpdateCommand {
         // Load / write SPX+ public key.
         let mut spx_private_key: Option<SpxSecretKey> = None;
         if let Some(key) = &self.spx_key {
-            let (pk, sk) = if let Ok(sk) = SpxSecretKey::read_pem_file(key) {
-                (SpxPublicKey::from(&sk), Some(sk))
-            } else {
-                (SpxPublicKey::read_pem_file(key)?, None)
+            let (pk, sk) = match SpxSecretKey::read_pem_file(key) {
+                Ok(sk) => (SpxPublicKey::from(&sk), Some(sk)),
+                Err(SpxError::Io(_)) => {
+                    // Return error if file could not be read.
+                    bail!(std::io::Error::new(
+                        std::io::ErrorKind::NotFound,
+                        "File not found"
+                    ));
+                }
```

Comment:
not sure if this is the most elegant way to do it, but this eliminates printing the same error twice when the key file does not exist

Reply:
We could choose to not try to print an error in the sphincsplus lib and rely on the higher level code printing the error.  To pick this call-site as an example, if we were to write something like this:
```
    let (pk, sk) = if let Ok(sk) = SpxSecretKey::read_pem_file(key) {
        (SpxPublicKey::from(&sk), Some(sk))
    } else {
        (SpxPublicKey::read_pem_file(key).with_context(|| format!("Failed to open {key:?}"))?, None)
}
```

Then the "File not found" error is going to display as:
```
Error:
  Failed to open 'foo.pem'
Caused by:
  File not found
```

Whereas some other error (like a PEM parsing error) will display as:
```
Error:
  Failed to open 'foo.pem'
Caused by:
  PEM parse error
```

Another possibility is to consider that `anyhow::Context` is implemented on the core rust error type.  I think you can use `Context` without using the rest of `anyhow`.  In that case, you could (in `sphincspluis/lib.rs`) write something like:

```
use anyhow::Context;
...
let s = std::fs::read_to_string(filename)
    .map_err(SpxError::Io)
    .with_context(format!("Failed to open {filename:?}"))?;
```

(I'm not exactly sure how the Context gets attached to the error; I haven't studied that portion of the `anyhow` crate in detail).

Reply:
I thought about removing error reporting from the SPHINCS library, but decided against it, because it could be invoked from multiple places, so each place would have to add error processing.

should this not be a concern?

Reply:
The `eprintln!` is unconditional - I much prefer error reporting through the Error trait rather than extraneous prints buried in a library (IMHO, that style of error reporting or runtime state should be handled by a logging library).

I think in this case, the two best possibilities are:
1. Add the context object at the library call-sites.
2. Add the context object in the sphincsplus library.

In both cases, the helpful message describing which file had trouble gets presented to the user when the program terminates and prints its final result.

Reply:
I am not really sure how to `Add the context object in the sphincsplus library`, may I suggest to leave this as a future exercise.

I improve the error reporting in `command/image.rs` so at least on manifest update invocations file name will be reported. 

Also added the fix to the bootstrap option - another place where I run into this error quite often.

PTAL

Reply:
sgtm